pkg/rbac/authenticationMiddleware_test.go

```diff
@@ -10,7 +10,13 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-//test valid token from cookie
+// Runs before the tests
+func TestMain(m *testing.M) {
+	// Replace the cache with a mock cache with a fake kubernetes client.
+	cache = newMockCache()
+}
```

Comment:
Test were using a real connection to the kubernetes cluster, which created inconsistent results.  This replaces the real connection with a mock client.

@Anxhela21 FYI, this resolves the problem of tests failing when running in prow.